packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -60,21 +60,47 @@ const NODE_MANIFEST_FILE_LIMIT = getNodeManifestFileLimit()
  */
 async function findPageOwnedByNode({
   nodeId,
+  fullNode,
   slug,
 }: {
   nodeId: string
+  fullNode: IGatsbyNode
   slug?: string
 }): Promise<{
   page: INodeManifestPage
   foundPageBy: FoundPageBy
 }> {
   const state = store.getState()
   const { pages, nodes } = state
-  const { byNode } = state.queries
-
-  // the default page path is the first page found in
-  // node id to page query tracking
-  let pagePath = byNode?.get(nodeId)?.values()?.next()?.value
+  const { byNode, byConnection, trackedComponents } = state.queries
+
+  const { staticQueryComponents } = state
+  const nodeType = fullNode?.internal?.type
+
+  let pagePath =
+    // the default page path is the first page found in
+    // node id to page query tracking
+    byNode?.get(nodeId)?.values()?.next()?.value ||
+    // otherwise the first page that queries for a list of this node type.
+    // we don't currently store a list of node ids for connection fields to queries
+    // we just store the query to connection types.
+    byConnection.get(nodeType)?.values()?.next()?.value
+
+  // for static queries, we can only find the first page using that static query
+  if (pagePath?.startsWith(`sq--`)) {
```

Comment:
how would `pagePath` start with `sq--`? Is `pagePath` actual pages?

Reply:
`byConnection` either returns a page path or a static query id.